### PR TITLE
 simplify launch_FV3LAM_wflow.sh

### PR DESCRIPTION
**Pull request by guoqing-noaa**
## DESCRIPTION OF CHANGES: 

- simplify launch_FV3LAM_wflow.sh
- hotfix on Init.sh

## TESTS CONDUCTED: 

- On machines/platforms:

- [ ] WCOSS2
- [x] Hera
- [ ] Orion
- [x] Jet

- Test cases: 

- [x] Non-DA engineering test
- [ ] DA engineering test
- [x] Other sample scripts:  Realtime runs

## ISSUE: 

- Fixes the issue(s) mentioned in #114

### CODE REVIEW

File: ush/generate_FV3LAM_wflow.sh [hunk 1]
```diff
@@ -597,12 +597,15 @@ $settings"
...
-ln -fs "${WFLOW_LAUNCH_SCRIPT_FP}" "$EXPTDIR"
+print_info_msg "Generating the workflow launch script
+${EXPTDIR}/${WFLOW_LAUNCH_SCRIPT_FN}"
+
+cat > ${EXPTDIR}/${WFLOW_LAUNCH_SCRIPT_FN} <<EOF
+#!/bin/bash
+module load rocoto
+rocotorun -w "${WFLOW_XML_FN}" -d "${WFLOW_XML_FN%.*}.db"
+EOF
+
```
Comment (chan-hoo) [inline ush/generate_FV3LAM_wflow.sh:608]: @guoqing-noaa, I understand that a huge log file is generated for a DA run by the launch script. The rocoto xml file for a DA run is incomplete. There are so many tasks unlaunched each cycle. Some useful capabilities of the launch script are not used in a DA run. However, this script is very useful for a non-DA run. So I suggest you to put an if-statement here like
```
if [ "${DO_NON_DA_RUN}" = "TRUE" ]; then
  (current link)
else
  (your new script)
fi

File: ush/Init.sh [hunk 2]
```diff
@@ -6,7 +6,7 @@ utildir="$basedir"/bash_utils
 # set the pre-commit hook
 echo -e "#!/bin/sh" > $basedir/../.git/hooks/pre-commit
 chmod +x $basedir/../.git/hooks/pre-commit
-echo -e "bash_utils/preCommit">>$basedir/../.git/hooks/pre-commit
+echo -e "ush/bash_utils/preCommit">>$basedir/../.git/hooks/pre-commit
```
Comment (chan-hoo) [inline ush/Init.sh:9]: @guoqing-noaa, thanks for this fix! I was looking for this.

File: ush/generate_FV3LAM_wflow.sh (hunk 1 above)
Comment (chan-hoo) [reply ush/generate_FV3LAM_wflow.sh:608]: @guoqing-noaa, in addition, `module load rocoto` doesn't work on WCOSS2. Please check `modulefiles/wflow_wcoss2.lua`.

**Comment by MatthewPyle-NOAA** [comment]
Is it worth splitting this into 2 PRs?  Seems like the Init.sh fix is something we want committed quickly, while the launch script work may need more work.

File: ush/generate_FV3LAM_wflow.sh (hunk 1 above)
Comment (chan-hoo) [reply ush/generate_FV3LAM_wflow.sh:608]: @guoqing-noaa, I am looking through the launch script. As I mentioned above, it has some useful functions can be used when a rocoto xml file is complete. So my new suggestion is: how about generating a new script with a new name without removing the current link? I'll update the launch script later.

**Comment by chan-hoo** [comment]
@MatthewPyle-NOAA, yes I agree with you. The fix is urgent, but we'll need more discussion for the launch script.

File: ush/generate_FV3LAM_wflow.sh (hunk 1 above)
Comment (guoqing-noaa) [reply ush/generate_FV3LAM_wflow.sh:608]: @chan-hoo Thanks for the suggestion. It is a good idea to use an alternate name for a simplified version. I will make the change soon.

Comment (guoqing-noaa) [reply ush/generate_FV3LAM_wflow.sh:608]: Regarding the heavy launch script, I have alternate thoughts:
1. Users can always run rocotostat to check task status manually. So running rocotostat twice in each crob job is not a must. If the cycledef spans a few months (for example, 4 months in my RTMA run situation), the log file will grow very big in size quickly.
2. var_defns.sh is a huge file with 2991 lines. Sourcing it in each cron job but only using 3 or 4 variables is not that efficient for a cron job.
3. I understand that rocotorun may fail to submit some tasks due to various issues. However, the `FV3LAM_wflow_*.log` file may already provide some useful information for debugging.  If an issue is transient, we may not need to or may not be able to address it. If an issue is persistent, we can always debug based on the current cycle runs.

Just my two cents. Thanks!

**Comment by guoqing-noaa** [comment]
@chan-hoo I made changes based on the feedback from you and Matt. 
The simplified version is now named "run_rocoto.sh" and the original launch script is still linked.

I don't have access to wcoss2 so I cannot test the changes related to wcoss2. Could you help me test it on wcoss2?
Thanks!

File: ush/generate_FV3LAM_wflow.sh [hunk 3]
```diff
@@ -603,6 +603,19 @@ workflow launch script (WFLOW_LAUNCH_SCRIPT_FP):
   EXPTDIR = \"${EXPTDIR}\"
   WFLOW_LAUNCH_SCRIPT_FP = \"${WFLOW_LAUNCH_SCRIPT_FP}\""
 ln -fs "${WFLOW_LAUNCH_SCRIPT_FP}" "$EXPTDIR"
+
+print_info_msg "Generating an alternate simple launch script
+${EXPTDIR}/run_rocoto.sh"
+
+echo "#!/bin/bash" > ${EXPTDIR}/run_rocoto.sh
+if [[ $MACHINE == "wcoss2" ]] ; then
```
Comment (chan-hoo) [inline ush/generate_FV3LAM_wflow.sh:611]: @guoqing-noaa, it should be uppercases: wcoss2 => WCOSS2

Comment (guoqing-noaa) [reply ush/generate_FV3LAM_wflow.sh]: @chan-hoo Thanks a lot for catching this.  
I changed it to:
`
if [[ "${MACHINE,,}" == "wcoss2" ]] ; then
`
Hope it is a safer way and you are okay with it.

**Comment by chan-hoo** [comment]
The non-DA eng test was completed successfully on wcoss2.

**Review by chan-hoo — APPROVED** [review]

**Pull request merged**